packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.mm

```diff
@@ -1016,6 +1006,43 @@ - (void)setGhostText:(NSString *)ghostText {
   self.backedTextInputView.ghostTextChanging = NO;
 }
 
+/**
+ * Attempts to remove the ghost text from a provided string given our current state.
+ * If `strict` mode is enabled, this method asserts if we don't find the ghost text in the expected location.
```

Comment:
Why do we want strict mode?

Reply:
The idea is being able to disable strict checks in cases where we may have already removed the ghost text. There's a comment explaining this in a place where we invoke this method, but this can definitely be worded better here.

Reply:
> The idea is being able to disable strict checks in cases where we may have already removed the ghost text.

But so if we already removed ghost text, _ghostText is nil. So you want to check for cases where we still *think* we have ghost text, but somehow we don't have all of it. And in that case, we clear the whole input string? (by returning nil)

That's the part I didn't understand -- if we couldn't remove the ghost text because it's partially gone (user managed to edit it?), aren't we making it worse by then clearing the text box?

Reply:
Returning nil doesn't immediately write nil to the text box; it's just an indication that an error occurred. If you look at the places where we call `removingGhostTextFromString:strict:`, we make sure that the value we get back is non-nil before we actually lock it in.

Reply:
Indeed. I missed that part.